Rails4 (don't merge, just review)

unorthodoxgeek opened the pull request:
this monstrous pull request is everything which is needed for rails 4.
dear courageous reviewer - may the odds be ever in your favor.

Commit:
fixes #7594 - extract params defined in apipie documentation to use for strong parameters

shlomizadok committed:
refs #3917 - rebase, added new permissions (mail_notification), lookup_keys.js

shlomizadok committed:
fixes #3917 - Added Strong Parameters

unorthodoxgeek committed:
refs #3157 - project still broken, but rails boots up

unorthodoxgeek committed:
fixes #8726 - recreating the hash_for method in the routes helpers

unorthodoxgeek committed:
fixes #8737 - removing deprecation warning due to has_many deprecation of conditions, whiny nils, etc. also, gems which have wrong dependencies now work from their github forks

Commit:
fixes #7567 - update syntax of deprecated find_by_[attribute] dynamic methods for rails 4

unorthodoxgeek committed:
fixes #8738 - tests are running, but failing

Commit:
fixes #7568 - remove attr_accessible and attr_protected for rails 4 upgrade

Commit:
fixes #3917 - change API controllers use to strong parameters based on Apipie documentation

unorthodoxgeek committed:
refs #3157 - downgrading to rails 4.0.2

dLobatog commented on Gemfile:
```diff
@@ -5,10 +5,10 @@ require File.expand_path('../lib/regexp_extensions', FOREMAN_GEMFILE)
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.21'
+gem 'rails', '4.0.12'
```
I think this should be `4.2`, or 4.1.9 at the very least. 4.0 is retired which means no bug fixes, no security fixes, etc... for this Rails version. Read the details [here](http://weblog.rubyonrails.org/2015/1/6/Rails-4-1-9-and-4-0-13-have-been-released/)

unorthodoxgeek replied:
rails versions should only be upgraded one minor at a time, we moved one minor from 3.2 to 4.0.

dLobatog replied:
This means we're moving to a version that gets security fixes to another that doesn't?

unorthodoxgeek replied:
as a first step, yes.

domcleal replied:
This is quite concerning, maybe it ought to be discussed on -dev?  Obviously, the available SCLs will be a factor too...

domcleal replied:
I checked and Rails 4.1.5 will be in a future SCL, so we can target that for a release.  I agree with @eLobato that 4.0 doesn't seem a possible _release_ target due to lack of security support - but we could upgrade to 4.0 and then to 4.1 before a release, if that's feasible.

dLobatog replied:
/me trembles. I don't see many other reasonable options, let's try to merge this one as early as possible after 1.8 so that we don't merge 4.1 right before 1.9

dLobatog commented on Gemfile:
```diff
@@ -19,12 +19,14 @@ gem 'oauth', '~> 0.4'
 gem 'gettext_i18n_rails', '~> 1.0'
 gem 'i18n', '~> 0.6.4'
+gem 'activerecord-session_store'
+gem 'rails-observers', git: 'https://github.com/rails/rails-observers.git'
```
No need to use the latest `rails-observers` from git master right? :smile:

dLobatog commented on app/helpers/strong_parameters_helper.rb:
```diff
@@ -0,0 +1,98 @@
+    Foreman::PermittedAttributes
+  end
+
+  def foreman_params
```
Why is this method `foreman_params` instead of `permitted_params`? It is not very descriptive unfortunately as it is currently.

dLobatog commented on app/helpers/strong_parameters_helper.rb:
```diff
@@ -0,0 +1,98 @@
+  end
+
+  def permitted_user_attributes
+    if editing_self?
```
Please use a guard clause instead of wrapping the whole method in an if else clause.

unorthodoxgeek replied:
what's a guard clause?

dLobatog commented on app/helpers/strong_parameters_helper.rb:
```diff
@@ -0,0 +1,98 @@
+
+  def vm_attributes
+    [
+        ## case ComputeResource.type ??
```
Try to make this method understand which provider attributes are needed, as the comment 'says'

dLobatog commented on app/models/hostgroup.rb:
```diff
@@ -91,6 +91,11 @@ class Jail < Safemode::Jail
 
   #TODO: add a method that returns the valid os for a hostgroup
 
+  def to_param
```
Isn't this method already provided by `Parameterizable::ByIdName`?

unorthodoxgeek replied:
the method in Parameterizable::ByIdName uses name, this one uses title.

dLobatog commented on config/initializers/active_record_extensions.rb:
```diff
@@ -4,3 +4,54 @@ class ActiveRecord::Base
 end
+
+
+#this fixes annoying issue with custom counter cache column names
```
What annoying issue?

unorthodoxgeek replied:
counters are incremented twice.
solution adapted from here: https://github.com/rails/rails/issues/10865

dLobatog commented:
[test]

domcleal commented:
It's failing consistently, doesn't need a re-test.

```
00:04:49.997 NameError: undefined method `it' for class `Class'
```

We were looking at this the other day - it was happening because the rake jenkins:unit task was running in the development environment when it parsed test_helper, while on develop/rails3 it's running in the test environment.  Maybe something needs changing in the task definition in lib/tasks/?

dLobatog commented on app/assets/javascripts/lookup_keys.js:
```diff
@@ -69,7 +69,7 @@ function add_child_node(item) {
     if (assoc == 'lookup_keys') {
       $('#smart_vars .smart-var-tabs .active, #smart_vars .stacked-content .active').removeClass('active');
-      var pill = "<li class='active'><a data-toggle='pill' href='#new_" + new_id + "' id='pill_new_" + new_id + "'><div class='clip'>" + __('new') + "</div><span class='close pull-right'>&times;</span></a></li>"
+      var pill = "<li class='active'><a data-toggle='pill' href='#" + new_id + "' id='pill_" + new_id + "'><div class='clip'>" + __('new') + "</div><span class='close pull-right'>&times;</span></a></li>"
```
Isn't this removal going to make it not match the RegExp in fix_template_names?

unorthodoxgeek replied:
honestly, I have no idea what this line is, I didn't put it there...

ohadlevy replied:
maybe all of the strong params needs to be merged separately from thsi branch? @shlomizadok @eLobato ?

unorthodoxgeek replied:
it will create more issues than it will solve.

On Thu, Apr 9, 2015 at 12:08 PM, Ohad Levy notifications@github.com wrote:

> In app/assets/javascripts/lookup_keys.js
> https://github.com/theforeman/foreman/pull/2055#discussion_r28046027:
> 
> > @@ -69,7 +69,7 @@ function add_child_node(item) {
> >      var field   = '';
> >      if (assoc == 'lookup_keys') {
> >        $('#smart_vars .smart-var-tabs .active, #smart_vars .stacked-content .active').removeClass('active');
> > -      var pill = "<li class='active'><a data-toggle='pill' href='#new_" + new_id + "' id='pill_new_" + new_id + "'><div class='clip'>" + __('new') + "</div><span class='close pull-right'>&times;</span></a></li>"
> > -      var pill = "<li class='active'><a data-toggle='pill' href='#" + new_id + "' id='pill_" + new_id + "'><div class='clip'>" + __('new') + "</div><span class='close pull-right'>&times;</span></a></li>"
> 
> maybe all of the strong params needs to be merged separately from thsi
> branch? @shlomizadok https://github.com/shlomizadok @eLobato
> https://github.com/eLobato ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theforeman/foreman/pull/2055/files#r28046027.

dLobatog commented on app/controllers/api/v2/host_classes_controller.rb:
```diff
@@ -36,7 +36,7 @@ def destroy
       def find_host
         not_found and return false if params[:host_id].blank?
-        @host = Host.find(params[:host_id]) if Host::Managed.respond_to?(:authorized) &&
+        @host = Host.friendly.find(params[:host_id]) if Host::Managed.respond_to?(:authorized) &&
```
Any reason not to use `find_resource`? `FindCommon` should handle authorization/scope logic everywhere

unorthodoxgeek replied:
the logic in host classes is slightly different from the logic in find common, I didn't change that, just added the friendly (from the horrible friendly_id gem)

dLobatog commented on app/controllers/api/v2/operatingsystems_controller.rb:
```diff
@@ -7,7 +7,7 @@ class OperatingsystemsController < V2::BaseController
 
       before_filter :find_optional_nested_object
-      before_filter :find_resource, :only => %w{show edit update destroy bootfiles}
+      before_filter :find_resource, :only => %w{edit update destroy bootfiles}
```
Why don't you let find_resource find the OS on show?

unorthodoxgeek replied:
the find method on Operatingsystem works differently from the rest of the models (it's find by id / name / description / fullname), when using find_resource show doesn't work like it's supposed to.

dLobatog commented on app/controllers/application_controller.rb:
```diff
@@ -148,7 +148,7 @@ def action_permission
     @resource_base ||= model_of_controller.respond_to?(:authorized) ?
         model_of_controller.authorized(current_permission) :
-        model_of_controller.scoped
+        model_of_controller.all
```
Why?

unorthodoxgeek replied:
scoped has been deprecated. all is the new scoped.

dLobatog commented on app/controllers/auth_source_ldaps_controller.rb:
```diff
@@ -2,7 +2,7 @@ class AuthSourceLdapsController < ApplicationController
 
   def index
-    @auth_source_ldaps = resource_base.all
+    @auth_source_ldaps = resource_base.load
```
Why eager load this?

unorthodoxgeek replied:
don't know. I just preserved the behaviour.

dLobatog commented on app/controllers/concerns/find_common.rb:
```diff
@@ -22,12 +22,18 @@ def resource_scope(options = {})
         scope
       end
+
+      if scope.respond_to?(:friendly)
```
Can we override the models instead to have this by default? If not, please add a test ensuring `resource_class.all.authorized(....).friendly` is not overridding the authorized scope.

unorthodoxgeek replied:
testing it would mean testing the scope mechanism, which is silly, as it's fairly well tested. scopes, by their definition, don't override each other.

dLobatog commented on app/controllers/concerns/foreman/controller/taxonomies_controller.rb:
```diff
@@ -197,4 +197,8 @@ def count_nil_hosts
     @count_nil_hosts = Host.where(taxonomy_id => nil).count
   end
 
+  def taxonomy_params
```
Is this ever used?

unorthodoxgeek replied:
nope. removing.

dLobatog commented on app/controllers/hosts_controller.rb:
```diff
@@ -159,11 +159,12 @@ def puppetclass_parameters
   #expected a fqdn parameter to provide hostname to lookup
   #see example script in extras directory
   #will return HTML error codes upon failure
+  #TODO: rename this to external_nodes
```
Rename it if you want :smile:

eLobato commented on app/helpers/bmc_helper.rb:
```diff
@@ -13,7 +13,7 @@ def power_actions
         display_link_if_authorized(_(action.to_s.capitalize),
                                    { :action => "power", :id => @host, :power_action => action, :auth_object => @host },
-                                   :confirm => _('Are you sure?'), :method => :put)
+                                   :data => { :confirm => _('Are you sure?') }, :method => :put)
```
I'm not sure why Rails 4 would make us do this..? Can you point to the bootstrap docs where this is explained?

dLobatog commented on app/helpers/strong_parameters_helper.rb:
```diff
@@ -0,0 +1,98 @@
+    end
+  end
+
+  def permitted_compute_attribute_attributes
```
Move `permitted_classname_attributes` to their respective controllers, this helper is currently keeping information very specific to the controllers.

dLobatog commented on app/models/bookmark.rb:
```diff
@@ -21,7 +18,7 @@ class Bookmark < ActiveRecord::Base
   scope :my_bookmarks, lambda {
     user = User.current
-    return {} unless SETTINGS[:login] and !user.nil?
+    return all unless SETTINGS[:login] and !user.nil?
```
Return **all** bookmarks? Seems reasonable if there is no login, but document it (an issue on Redmine would work) Change to user.present?, double negative is confusing.

unorthodoxgeek replied:
"all" is just an empty scope. scopes can no longer return a hash, they must return a different scope. can be changed to where({}), but I prefer all

dLobatog commented on app/models/concerns/parameterizable.rb:
```diff
@@ -13,7 +13,15 @@ def to_param
 
       def self.from_param(id)
-        self.find(id.to_i)
+        self.where(id: id.to_i).first
```
I like this syntax, but let's be consistent with the rest of the code base & hash rockets

dLobatog commented on app/models/concerns/parameterizable.rb:
```diff
@@ -46,7 +62,15 @@ def to_param
+        self.where(name: name).first
+      end
+
+      def self.xfind(*args)
```
`xfind` typo? 

This is exactly the same to the method used in ById

dLobatog commented on app/models/concerns/parameterizable.rb:
```diff
@@ -28,7 +36,15 @@ def to_param
+      end
+
+      def self.find(*args)
+        if args.size == 1 && (args.first.is_a?(String) || args.first.is_a?(Numeric))
```
Can we refactor all the .find methods to just one? They only differ (slightly!) in the conditional.

unorthodoxgeek replied:
will do.

dLobatog commented on app/models/taxable_taxonomy.rb:
```diff
@@ -6,7 +6,7 @@ class TaxableTaxonomy < ActiveRecord::Base
 
   scope :without, lambda{ |types|
     if types.empty?
```
condition = types.present? ? ["taxable_taxonomies.taxable_type NOT IN (?)",types] : {}
where(condition)

unorthodoxgeek replied:
I strongly dislike ternary operators which have a line longer than 10
characters.

dLobatog commented on app/models/host/managed.rb:
```diff
@@ -65,17 +65,17 @@ class Jail < ::Safemode::Jail
-
-  scope :recent,      lambda { |*args| {:conditions => ["last_report > ?", (args.first || (Setting[:puppet_interval] + 5).minutes.ago)]} }
-  scope :out_of_sync, lambda { |*args| {:conditions => ["last_report < ? and enabled != ?", (args.first || (Setting[:puppet_interval] + 5).minutes.ago), false]} }
+  # default_scope lambda {
```
Why the commented code?

unorthodoxgeek replied:
that's a great question - I don't know why this breaks everything. it just
does. that's why I commented it out instead of deleting...

dLobatog commented:
@unorthodoxgeek Still haven't reviewed it very thoroughly, let me know when the above is fixed and I'll spare one complete day to fully review and work on this with you.

unorthodoxgeek committed:
fixes #1881 - adding auto refresh button to dashboard

unorthodoxgeek committed:
fixes #9551 - checking the encryption format so osx doesn't fail

unorthodoxgeek committed:
refs #3157 - fix mailer tests

unorthodoxgeek committed:
fixes #10232 - moving validations and casting out of lookup key and value

unorthodoxgeek committed:
fixes #10233 - removing DatabaseCleaner from the after-runs

Commit:
Fixes #10248 - cloning volume information for virtual machines

dLobatog commented on app/assets/javascripts/application.js:
```diff
@@ -112,6 +112,11 @@ function onContentLoad(){
   $.cookie('timezone', tz.name(), { path: '/' });
+
+  // Removes all the templates from forms as they can't be cleared by strong parameters.
+  $('form').on('submit', function(e) {
```
This only happens when editing Host interfaces, and Lookup keys right? Can it be moved there instead? Seems like a waste of time to do it everywhere.

ohadlevy replied:
just include it in the same place we have the lookup_keys js?

On Tue, May 5, 2015 at 2:47 PM, Daniel Lobato García <
notifications@github.com> wrote:

> In app/assets/javascripts/application.js
> https://github.com/theforeman/foreman/pull/2055#discussion_r29661548:
> 
> > @@ -112,6 +112,11 @@ function onContentLoad(){
> > 
> >    var tz = jstz.determine();
> >    $.cookie('timezone', tz.name(), { path: '/' });
> > +
> > -  // Removes all the templates from forms as they can't be cleared by strong parameters.
> > -  $('form').on('submit', function(e) {
> 
> This only happens when editing Host interfaces, and Lookup keys right? Can
> it be moved there instead? Seems like a waste of time to do it everywhere.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theforeman/foreman/pull/2055/files#r29661548.

dLobatog commented on app/controllers/api/v1/puppetclasses_controller.rb:
```diff
@@ -16,8 +16,8 @@ def index
           select([:name, :id]).
-          includes(:lookup_keys)
-        render :json => Puppetclass.classes2hash(values.all)
+          includes(:lookup_keys).to_a
```
Curious about this `.to_a` after most includes?

unorthodoxgeek replied:
in AR 4.1 the #all method has been changed. to_a does it. this is part of
the deprecation warnings cleanup, I think...

dLobatog replied:
I get it but I was wondering if we can avoid eager loading this on the API and wait until it's rendered?

dLobatog commented on app/controllers/api/v2/interfaces_controller.rb:
```diff
@@ -6,6 +6,8 @@ class InterfacesController < V2::BaseController
       include Api::Version2
       include Api::TaxonomyScope
 
+      LEGACY_TYPE_NAMES = Nic::Base.allowed_types.map{ |t| t.name }
```
This is never used here? It seems like it was copied from interface_type_mapper but here it's not used.

dLobatog commented on app/services/authorizer.rb:
```diff
@@ -67,6 +68,7 @@ def build_filtered_scope_components(resource_class, all_filters, options)
     end
 
     return result if all_filters.any?(&:unlimited?)
+    # return resource_class.all if all_filters.any?(&:unlimited?)
```
Why new commented code?

dLobatog commented on app/models/concerns/counter_cache_fix.rb:
```diff
@@ -7,7 +7,7 @@ module CounterCacheFix
 
   included do
-    after_update :update_counter_caches
+    #after_update :update_counter_caches
```
?

dLobatog commented on app/models/architecture.rb:
```diff
@@ -1,7 +1,5 @@
 class Architecture < ActiveRecord::Base
   include Authorizable
-  extend FriendlyId
```
Why?

dLobatog commented:
@unorthodoxgeek Could you create an associated issue to Rails 4 in Redmine and move these files there?

I think that should be ready to merge, and it's compatible with Rails 3.

config/routes.rb
config/routes/api/v1
config/routes/api/v2
db/migrate/*
app/views/users/_form
app/views/filters/_form
app/views/config_groups/_config_groups_selection.html.erb
app/views/compute_resources/show.html.erb
app/views/common/_edit_habtm.html.erb
app/services/puppet_fact_parser.rb
app/services/puppet_fact_importer.rb
app/services/classification/base.rb
app/services/authorizer.rb
app/models/taxonomy.rb
app/models/report.rb
app/models/concerns/authorizable.rb
app/models/compute_resources/foreman/model/vmware.rb

dLobatog commented:
https://github.com/theforeman/foreman/pull/2358

Commit:
refs #3157 - Reference all tables we might be referring to in the search query

orrabin committed:
Fixes #4419 - rearranging smart class parameters edit form

dLobatog commented:
@unorthodoxgeek This needs to be updated after the last inclusion of some Rails 3 compatible changes (now I'm looking for more of these)

unorthodoxgeek commented:
Can you please rebase your update into this branch?
On Jun 17, 2015 8:23 PM, "Daniel Lobato García" notifications@github.com
wrote:

> @unorthodoxgeek https://github.com/unorthodoxgeek This needs to be
> updated after the last inclusion of some Rails 3 compatible changes (now
> I'm looking for more of these)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theforeman/foreman/pull/2055#issuecomment-112883524.

dLobatog commented:
@unorthodoxgeek I rebased the entirety of develop, including the ConfigTemplate > ProvisioningTemplate change which rendered a lot of this PR outdated here https://github.com/eLobato/foreman/pull/8 (didn't want to open a new PR). I'll look what parts can I keep on extracting and make compatible with Rails 3, there must be room to make it smaller - specifically I'm looking at foreman_params and the cattr_accessor change in Nic.

Why is friendly_id not extended in most models now?

unorthodoxgeek commented:
@eLobato please see the rails4-develop branch on my repo (I've fixed some code so db:migrate works and project boots up) - tests are not running on it for some reason, can you take a look?

dLobatog commented:
I'm keeping track of how far are we from develop [here](https://github.com/eLobato/foreman/pull/8/files).
This comment will keep track of what we can and cannot submit separately:

_Can_
- strong_params. Move permitted_attributes to their own classes instead of a central lib/foreman/permitted_attributes - https://github.com/theforeman/foreman/pull/2509
- `:data => { :confirm` deprecation - https://github.com/theforeman/foreman/pull/2476
- `.all` to `where xxx .all`
- `.find_all_by`. to `where(:attribute =>` https://github.com/theforeman/foreman/pull/2533
- `random_macs` and `random_ip` factories
- `valid_attrs` changes to tests.
- wrap scope relationships in lambdas - https://github.com/theforeman/foreman/pull/2605 , only log.rb 

_Cannot_
- friendly_id 5.0 
- `find_or_create_by`

domcleal commented on Gemfile:
```diff
@@ -5,13 +5,13 @@ require File.expand_path('../lib/regexp_extensions', FOREMAN_GEMFILE)
 gem 'will_paginate', '~> 3.0'
 gem 'ancestry', '~> 2.0'
-gem 'scoped_search', '~> 3.0'
+gem 'scoped_search', '~> 2.7'
```
Should be able to revert this, as 3.2.1 fixes the last known Rails 4 issue.

unorthodoxgeek commented:
squashed a bunch of commits, mainly small commits which fix tests

Commit:
refs #3157 - revert "fixes #10627 - apply host taxonomy scope to facts/report joins"

Rails 4 correctly applies default scopes even when doing complex joins,
so the `Report.joins(:host => [:hostgroup])` join would apply report,
host and hostgroup default scopes.

Both host and hostgroup have taxonomy-based default scopes which is
enough to resolve the original bug.

The extra change to the tests to set hostgroup taxonomies is because
the original fix only applied the host taxonomy scope and not hostgroup,
so this is now required to avoid mismatches.

This reverts commit 9926db4274e2d9d01466676116a9c001f473f1a4.

unorthodoxgeek commented:
yoohoo! down to 15 commits without the robot shouting about them :smile:

avi3tal committed:
fixes #11220 - column ambiguity for non-admin host creation

Commit:
refs #3157 - Fix frozen array issues

dLobatog commented:
I rebased this branch with develop today, found some breakages. Take a sneak peak at [my repo](https://github.com/eLobato/foreman/pull/9/), the diff has narrowed its scope significantly I think. At this point it's mostly `find_or_create_by` finders, `strong_params` and `friendly_id` that compose the majority of the changes.

Commit:
fixes #11170 - pass quoted args through foreman-rake

Use the special "$@" to correctly pass quoted arguments through
foreman-rake, through su and to the underlying rake command.  Ensures
the JSON in this command is unharmed:

    foreman-rake config -- -k trusted_puppetmaster_hosts -v '["foo.bar.com"]'

Also swallows the "--" which gets passed through on some versions of
rake.

slivik committed:
fixes #11274 - Unable to get resourcePoo

Cluster attribute full_path makes sense when there are intermediate folders (both between vCenter/datacenter and datacenter/cluster). There was a bug/typo in fog (https://github.com/slivik/fog/pull/1) and after it is merged this should work.

ares committed:
Fixes #11187 - explicitly pass resource_type and permission to check

Commit:
fixes #11312 - Allow plugins to update the Rabl view_paths

amirfefer committed:
Fixes #11174 - Don't show password placeholder when empty

ShimShtein committed:
Fixes #11255 - Fixed inheritance checks for explicit input on host clone

jlsherrill committed:
Fixes #11339 - ensure unique parameters before adding uniq index

Commit:
fixes #10997 - update fog to 1.33.0

komidore64 commented:
It looks like we may also need to alter [ActiveRecord scopes](http://guides.rubyonrails.org/active_record_querying.html#scopes) by wrapping the relations with lambdas.

from

``` ruby
scope :has_url, where('url IS NOT NULL')
```

to

``` ruby
scope :has_url, lambda { where('url IS NOT NULL') }
```

I was getting some `undefined method 'call' for #<ActiveRecord::Relation []>` errors during migrations that lead me to seeing this in the RoR guides. It appears that Foreman already has a fair number of scopes wrapped in lambdas but it's probably worth specifically looking for them to find any that may not be wrapped.

This is a change that Katello can make in the meantime which won't break rails3 compatibility.

dLobatog commented:
@komidore64 Thanks for the heads up, will follow up with a PR to foreman with all of the scopes wrapped, if there are any after https://github.com/theforeman/foreman/pull/1787

shlomizadok committed:
fixes #11240 - Add api docs to hostgroup params

shlomizadok committed:
fixes #11374 - re-initialize select2 on smart vars

amirfefer committed:
Fixes #11371,#4410 - puppet parameters truncated and tooltip added

tbrisker committed:
Fixes #11376 - show location tab in organization editor

alongoldboim committed:
Fixes #11372 - Formalize assets names.

Commit:
fixes #11385 - Enable management of nics in compute profiles without libvirt network

unorthodoxgeek committed:
refs #3157 - fixing some stuff, wishing for death

unorthodoxgeek committed:
refs #11425 - adding lookup_value_matcher to host and hostgroup to make db schema more logical

Conflicts:
	app/models/concerns/host_common.rb

unorthodoxgeek committed:
refs #11425 - adding lookup_value_matcher to host and hostgroup to make db schema more logical

Conflicts:
	app/models/concerns/host_common.rb
	db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb
	test/unit/host_config_group_test.rb

tbrisker committed:
Fixes #11432 - Allow copying from ACE editor

unorthodoxgeek committed:
Merge branch 'develop' of github.com:theforeman/foreman into rails4

Conflicts:
	bundler.d/assets.rb

unorthodoxgeek commented:
[test]

unorthodoxgeek committed:
refs #3157 - fixing tests

dLobatog commented:
https://github.com/theforeman/foreman/pull/2618 takes the lookup_value_matcher migration in this PR and puts it in develop. Nice!

komidore64 committed:
fixes #11590 - include foreman's permitted_attributes

unorthodoxgeek committed:
Merge pull request #13 from komidore64/rails4-permitted-attributes

fixes #11590 - include foreman's permitted_attributes

dLobatog commented on test/unit/host_test.rb:
```diff
@@ -219,7 +219,8 @@ def teardown
 
-  test "should be able to delete existing lookup value on update_attributes" do
+  #test "should be able to delete existing lookup value on update_attributes" do
+  def test_fuck
```
huh?

unorthodoxgeek committed:
merge with develop

unorthodoxgeek committed:
refs #3157 - deleting silly comment

unorthodoxgeek committed:
refs #3157 - fix silly error
